### ISSUE TITLE
Fails to work with Raspbian Jessie

### ISSUE THREAD

**Issue by Paul-Reed** [adafruit/Adafruit-Pi-ExternalRoot-Helper#4]
Works great with Raspbian Wheezy, but just tried the script with Jessie, and after installing gdisk, rsync & parted the script stops, & displays;  
`[fs create] Creating /dev/sda1`
`Error: Partition(s) on /dev/sda are being used.`

The powered USB HDD has just one freshly formatted ext4 partition (no data), and has never been mounted on the Raspberry Pi before.

`pi@raspberrypi ~ $ df -h`
`Filesystem      Size  Used Avail Use% Mounted on`
`/dev/root       7.0G  3.3G  3.4G  50% /`
`devtmpfs        214M     0  214M   0% /dev`
`tmpfs           218M     0  218M   0% /dev/shm`
`tmpfs           218M  4.6M  213M   3% /run`
`tmpfs           5.0M  4.0K  5.0M   1% /run/lock`
`tmpfs           218M     0  218M   0% /sys/fs/cgroup`
`/dev/mmcblk0p1   56M   20M   37M  36% /boot`
`tmpfs            44M     0   44M   0% /run/user/1000`
`/dev/sda1       932G   94M  932G   1% /media/pi/087F97566F198312`

Paul

**Comment by brennen** [comment]
Thanks for the bug report - I used it recently with a fresh Jessie install and (somewhat to my surprise) it worked without any hitches, but I can well imagine that might not be the case for every system.

Actually, come to think of it, I'm pretty sure I know what the problem is - something in the desktop environment (which starts at boot now by default) is almost certainly auto-mounting the drive, and that's getting in the way of `parted`.  You could try `sudo umount /dev/sda1` (and maybe, for good measure, exit out of the desktop so the filemanager isn't running).  I'll work up a little patch to handle this situation more gracefully.

**Comment by Paul-Reed** [comment]
Thanks for the quick reply.
I've just run $ sudo raspi-config and changed the boot option to command prompt instead of desktop, rebooted and your script now works fine!
Thanks

Paul

**Comment by brennen** [comment]
Right on - glad to hear it's working.  I'll leave this issue open 'til I get a new version of the package uploaded with at least a warning in it.

**Comment by Paul-Reed** [comment]
A minor issue, but once root has been successfully moved to the HDD:
`$ readlink /dev/root`
does not return a value as per the guide, yet:
`$ ls -l /dev/disk/by-label`
shows that the change has occured.

Paul

**Comment by brennen** [comment]
Hrm. I'll check what's changed there.

**Comment by spambus** [comment]
I'm using jessie, is it possible that it takes hours to complete this script on a 1TB drive?

the script is bussy for hours allready and stuck on this part: [fs create] Creating /dev/sda1

when I look at the code it is either exexcuting this:

parted --script "${target_drive}" mklabel gpt

ot this:

parted --script --align optimal "${target_drive}" mkpart primary ext4 0% 100%

is it possible the "--align optimal" part takes hours on a 1TB drive?

**Comment by Paul-Reed** [comment]
No, it shouldn't take that long. Are you sure that your'e not booting Jessie into dashboard rather than command prompt? If the dashboard loads it automounts the HDD, and halts the process.

**Comment by spambus** [comment]
My pi only boots to terminal, and I run the script through ssh, I don't use desktop on my pi

**Comment by spambus** [comment]
Also, I had the usb hdd allready formatted to ext-4 with a usb stick with G-Parted on a windows computer, so basicly the script is doing, what I allready did before...
